### ISSUE TITLE
pyright reports spurious errors when using `attrs` validation decorators

### ISSUE THREAD

**Issue by thetorpedodog** [microsoft/pyright#4330]
When using `attrs` fields and the `@some_field.validator` decorators, pyright incorrectly marks use of `validator` as erroneous.

### To Reproduce

1. Ensure `attrs` is installed in your environment.
2. Use some level of type-checking via Pylance in VSCode above “off”.
3. Write an attrs class that uses the `@some_field.validator` decorator.

```python
import attrs


@attrs.define(frozen=True)
class Whatever:

    x: int = attrs.field()

    @x.validator
    def _validate(self, _, __):
        pass
```

### Expected behavior

My code passes pylance/pyright’s type-checking.

### Actual behavior

Under “basic” type-checking, Pylance reports on <code>@x.<b>validator</b></code>:

```
validator: Unknown
Cannot access member "validator" for type "int"
  Member "validator" is unknownPylance[reportGeneralTypeIssues](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportGeneralTypeIssues)
```

Under “strict” type-checking, Pylance complains harder (on <code>@<b>x.validator</b></code>):

```
validator: Unknown
Type of "validator" is unknownPylance[reportUnknownMemberType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportUnknownMemberType)
Untyped function decorator obscures type of function; ignoring decoratorPylance[reportUntypedFunctionDecorator](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportUntypedFunctionDecorator)
Cannot access member "validator" for type "int"
  Member "validator" is unknownPylance[reportGeneralTypeIssues](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportGeneralTypeIssues)
```

### Environment

I am using pyright via **Pylance v2022.12.20** in VS Code.

**Comment by erictraut** [comment]
Attrs implements behaviors that are unique to that library and cannot be described using the Python static type system. [PEP 681](https://peps.python.org/pep-0681/) was introduced in Python 3.11 to provide support for a subset of standardized dataclass-like behaviors implemented by attrs, pydantic, and other libraries, but it doesn't cover the non-standard (library-specific) behaviors. Pyright fully supports PEP 681.

If you want to use attrs with pyright or other standards-compliant type checker, you'll need to either avoid using these non-standard features of attrs or find some other workaround such as the use of `# type: ignore` comments to suppress the resulting type validation errors. I don't have much expertise with attrs, so I can't offer any more specific advice on how to work around the `validator` issue specifically. You may want to ask for further advice in the attrs discussion forum.
